scanpipe/pipes/d2d.py

```diff
@@ -953,3 +953,32 @@ def _map_javascript_colocation_resource(
         from_neighboring_resources,
         map_type="js_colocation",
     )
+
+
+def flag_processed_archives(project):
+    """
+    Resources without an assigned status which are package archives, and all
+    resources inside the archive has a status, should also be considered as
+    processed.
+    """
+    to_resources = project.codebaseresources.files().to_codebase()
+    to_resources_archives = to_resources.no_status().filter(is_archive=True)
+
+    for to_archive in to_resources_archives:
+        archive_extract_path = to_archive.path + "-extract"
+        archive_extract_resource = to_resources.filter(path=archive_extract_path)
+
+        # TODO:There are archives which are not extracted by default, these need to
+        # be accounted for. See https://github.com/nexB/scancode.io/issues/827
+        if not archive_extract_resource:
+            continue
```

Comment:
Let's not leave TODOs in the code, those never get addressed. We want issues instead, but here it seems that we have one.

Reply:
Using `archive_extract_resource.exists()` would result in a more efficient query.

Reply:
Thanks, removing the TODO there and using the exists() function.